core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -1143,7 +1135,6 @@ public AuthorizationManager getAuthorizationManager() {
 
    @Override
    public AdvancedCache<K, V> lockAs(Object lockOwner) {
-      Objects.nonNull(lockOwner);
```

Comment:
Why? DecoratedCache does not do the null check.

Reply:
`nonNull()` is a no-op since it only returns `true` if the `lockOwner` is `null`. No logic changes here.

However, I'm not sure if it can or not be null, but the `DecoratedCache` always check the field for `null`. Example:
```
if (lockOwner != null) {
     ctx.setLockOwner(lockOwner);
}
```

Reply:
Eh, right, I've mistaken it for `Objects.requireNonNull`. It seems that calling this with `null` parameter is clearly unintentional, the decorated cache does not decorate anything, then.

Reply:
ok. adding requireNonNull